docs/store.md

```diff
@@ -0,0 +1,42 @@
+# Pepr Store: A Lightweight Key-Value Store for Pepr Modules
+
+The nature of admission controllers and general watch operations (the `Mutate`, `Validate` and `Watch` actions in Pepr) make some types of complex and long-running operations difficult. There are also times when you need to share data between different actions. While you could manually create your own K8s resources and manager their cleanup, this can be very hard to manage or keep performant at scale. 
+
+The Pepr Store solves this by exposing a simple API compatible with the standard [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Storage) for use within capabilities that abstracts away the complexity of managing the data in K8s. Additionally, as Pepr runs multiple replicas of the admission controller and a single watch controller, the Pepr Store provides a unique way to share data between these different instances automatically.
+
+Each Pepr Capability has a `Store` instance that can be used to get, set and delete data as well as subsribe to any changes to the Store. Behind the scenes, all capability stores instances in a single Pepr Module are stored within a single CRD in the cluster. This CRD is automatically created when the Pepr Module is deployed. Care is taken to make the read and write operations as efficient as possible by using K8s watches, batch processing and patch operations for writes.
```

Comment:
"Each Pepr Capability has a `Store` instance that can be used to get, set and delete *arbitrary* data as well as *subscribe* to any changes to the Store."

Explanatory & typo.

Reply:
"Behind the scenes, all capability stores instances in a single Pepr Module are stored within a single CRD in the cluster. "

Not 100% on what this is saying.  Something about "all capability stores instances in a single Pepr Module are stored" feels like the plurality/subject aren't clear.

Reply:
"Care is taken to make the read and write operations as efficient as..."

Something about "Care is taken" feels tonally off -- something like "Focus has been put on making the read and write operations... etc." would read better.

Reply:
Yeah this whole thing might just need to be rewritten. Probably should have just had chatgpt write it instead of me 😆